Automattic/wp-calypso
# Domains: Create 100-Year Domain flow

## Proposed Changes

This diff creates the 100-Year Domain flow as described in p9Jlb4-e6E-p2.

In this flow, users are able to search and choose a .com/net/org domain.

Notes:
- This PR hardcodes the $2000 price for 100-year domains since we still don't haven't implemented the actual product
- For the same reason, checkout will also use the domains' regular prices instead of $2000

### Screenshots

- Initial step
![Screenshot 2024-10-10 at 19 18 53](https://github.com/user-attachments/assets/443cb382-5eff-4f89-94d3-f8353cfe66d4)
- Checkout
![Markup on 2024-10-10 at 19:20:11](https://github.com/user-attachments/assets/faa1bd19-0422-481c-af35-f40d0168ce3b)

## Why are these changes being made?

We want to offer 100-year domain registrations as a separate offer from the 100-year plan. More details in p9Jlb4-e6E-p2.

## Testing Instructions

- Build this branch locally or open the live Calypso link
- Enable `USE_STORE_SANDBOX` mode in your backend to prevent actual purchases
- Visit the flow at `/setup/hundred-year-domain`

Try to test the flow as thoroughly as possible:
- Complete the flow normally and ensure everything seems ok
  - Ensure that, after the flow is complete, the domain gets a domain-only site
- Test both in mobile and desktop layouts
- Try to hit "Back" in your browser and ensure the flow doesn't break
- Remove the domain from the shopping cart in the checkout page and ensure you go back to the domain search step
- Start the flow logged-in and logged-out
  - If you're logged-out, you should be able to search and select a domain before having to log in
- Ensure the domain search only returns .com/net/org domains

## Pre-merge Checklist
